apps/course-api/src/middlewares/caching.ts

```diff
@@ -10,6 +10,10 @@ export default async function cacheEndpoint(req: Request, res: Response, next: N
 	}
 
 	const redisClient = await getRedisClient();
+	if (!redisClient) {
+		return next();
+	}
+
 	const cachedResponse = await redisClient?.get(req.originalUrl);
```

Comment:
Okay, SO:

Optional chaining is used to access `get` only if `redisClient` is not `null` or `undefined`. The `await` is correctly placed to wait for the promise returned by get to resolve, but if the `redisClient` was `null` or `undefined`, the whole expression just  returns undefined immediately without any error, and the `await` has no effect since there's no promise to wait on.

Reply:
Not inherently "breaking" or anything, but something to keep in mind and avoid.

Something like this would be better because it's more explicit:
```typescript
let cachedResponse;

if (redisClient) {
	cachedResponse = await redisClient.get(req.originalUrl);
} else {
	// Handle the case where the Redis client is not available with logging
	console.error('Redis client is not available');
}
```

Reply:
Now that I think about it... I shouldn't need the `?` anyways since it'll simply be picked off by the if statement above.